main.go

```diff
@@ -42,6 +50,7 @@ func generateQRCodeHandler(w http.ResponseWriter, r *http.Request) {
 
     qr, err := qrcode.New(text, qrcode.Medium)
     if err != nil {
+        log.Printf("Failed to generating QR Code: %v", err)
```

Comment:
Good to add logging here

One normally also adds a middleware that logs the status and response of every request.